src/contracts/atlas/AtlasVerification.sol

```diff
@@ -548,27 +549,21 @@ contract AtlasVerification is EIP712, DAppIntegration, AtlasConstants {
     function _verifyUser(
         DAppConfig memory dConfig,
         UserOperation calldata userOp,
+        bytes32 userOpHash,
         address msgSender,
         bool isSimulation
     )
         internal
         returns (ValidCallsResult)
     {
+        if (userOp.from == address(this) || userOp.from == ATLAS || userOp.from == userOp.control) {
+            return ValidCallsResult.UserFromInvalid;
+        }
+
         // Verify the signature before storing any data to avoid
         // spoof transactions clogging up dapp userNonces
 
-        bool isFromContract = userOp.from.code.length > 0;
-
-        bool signatureValid = false;
-        if (isFromContract) {
-            if (userOp.from == address(this) || userOp.from == ATLAS || userOp.from == userOp.control) {
-                return ValidCallsResult.UserFromInvalid;
-            }
-            signatureValid = IAccount(userOp.from).validateUserOp(userOp, _getUserOpHash(userOp), 0) == 0;
```

Comment:
Can we get rid of `IAccount` interface definition (at the bottom of this file) then?

Reply:
Good catch, I was really looking forward to deleting this then totally forgot 😂